### ISSUE TITLE
Intervention Image Integration

### ISSUE THREAD

**Issue by frasmage** [plank/laravel-mediable#69]
Planning to add support for [Intervention](http://image.intervention.io/) manipulations while uploading images.

Currently considering an API like the following:

```php
MediaUploader::fromSource('example.jpg')
    ->applyManipulations(function(Image $image) {
         $image->resize(300, null);
    }, $quality = null)
    ->upload()

**Comment by jaesung2061** [comment]
Hey @frasmage , what is the progress on this? Will this be released soon?

**Comment by frasmage** [comment]
Hi @jaesung2061,

Glad to hear about your interest in this project!

Admittedly, between a couple of major changes in my life (for the better) in the last few months, I haven't had as much spare time to devote to working on the next version of this project as I would have liked. 

I will try to commit myself to cleaning some of my work-in-progress and push it to a public branch so that other can help move it towards the 3.0 goal. As always, pull requests are always welcome!

**Comment by sargbah84** [comment]
Really great media management package for Laravel. I'm really begging you to add the option for one to be able to generate different sizes from a single upload. One of my application requires uploading two different thumbnail images per a single image upload. I was really excited installing and playing around with your plugin in that application until I realize, there is no mention of resizing media in the documentation.

**Comment by leouze** [comment]
+1 Would love to have this as a feature as well.

**Comment by zefexdeveloper** [comment]
Is there any news on that or how to accomplish that? @frasmage

**Comment by simonschaufi** [comment]
I would also appreciate that feature. Shouldn't be too complicated :)

**Comment by ilyaskazi** [comment]
+ 1 from me... now we are in 2020 still waiting on this?

**Comment by frasmage** [comment]
Never got around to this particular feature. Until I get a chance to really explore this,  I'm open to PRs exploring how we might best implement this.

**Comment by M-Abdullahi** [comment]
There is this similar [package](https://github.com/optixsolutions/laravel-media) that I've discovered recently which shares similar features and includes media manipulations using  intervention package. Maybe we can port it to this package.

**Comment by frasmage** [comment]
Hey everyone! I have been working away for the last week trying to implement integrated support for the integration/image library. I think I am finally happy with the result! Here are the highlights:

* Define `ImageManipulations` which define what changes to make to file using intervention/image, and where to place the resulting file
* Create any number of "variants" of a `Media` record by applying the `ImageManipulations`. Can be done either synchronously in the current process or asynchronously as a queued job.
* Variants are associated to the original `Media` record, so one can easily be looked up from the other.

The changes are now available in the `5.0.0` branch (`dev-5.0.0` for composer version), draft documentation of the feature is available [here](https://laravel-mediable.readthedocs.io/en/5.0.0/variants.html). If anyone is interesting in testing out this feature over the next few days, I would love feedback before I release this.

**Comment by frasmage** [comment]
This feature has been released as part of 5.0.0
